product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/testng/listeners/TestExecutionListener.java

```diff
@@ -0,0 +1,237 @@
+                        .until(isCAppDeployed(applicationAdminClient, carFileName));
+
+            } else {
+                log.info("Carbon application : " + carFileName + " already deployed");
+            }
+        }
+    }
+
+    /**
+     * Function to undeploy carbon application
+     *
+     * @param applicationName
+     * @throws ApplicationAdminExceptionException
+     * @throws RemoteException
+     */
+    private void undeployCarbonApplication(String applicationName)
+            throws ApplicationAdminExceptionException, RemoteException {
+        if (standaloneMode) {
+            applicationAdminClient.deleteApplication(applicationName);
+
+            // Wait for Capp to undeploy
+            Awaitility.await()
+                    .pollInterval(500, TimeUnit.MILLISECONDS)
+                    .atMost(ScenarioConstants.ARTIFACT_DEPLOYMENT_WAIT_TIME_MS, TimeUnit.MILLISECONDS)
+                    .until(isCAppUnDeployed(applicationAdminClient, applicationName));
+        }
+    }
+
+    private void setKeyStoreProperties() {
+        System.setProperty("javax.net.ssl.trustStore", resourceLocation + "/keystores/wso2carbon.jks");
+        System.setProperty("javax.net.ssl.trustStorePassword", "wso2carbon");
+        System.setProperty("javax.net.ssl.trustStoreType", "JKS");
+    }
+
+    private String getServerHost(String url) {
+        if (url != null && url.contains("/")) {
+            url = url.split("/")[2].split(":")[0];
+        } else if (url == null) {
+            url = "localhost";
+        }
+        log.info("Backend URL is set as : " + url);
+        return url;
+    }
+
+    private Callable<Boolean> isCAppDeployed(final ApplicationAdminClient applicationAdminClient, final String cAppName) {
+        return new Callable<Boolean>() {
+            @Override
+            public Boolean call() throws Exception {
+
+                log.info("Check CApp deployment : " + cAppName);
+                return checkCAppDeployed(applicationAdminClient, cAppName);
+            }
+        };
+    }
+
+    private boolean checkCAppDeployed(ApplicationAdminClient applicationAdminClient, final String cAppName)
+            throws ApplicationAdminExceptionException, RemoteException {
+        log.info("Check CApp deployment : " + cAppName);
+        String[] applicationList = applicationAdminClient.listAllApplications();
+        if (applicationList != null) {
+            for (String app : applicationList) {
+                if (app.equals(cAppName)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    private Callable<Boolean> isCAppUnDeployed(final ApplicationAdminClient applicationAdminClient, final String cAppName) {
+        return new Callable<Boolean>() {
+            @Override
+            public Boolean call() throws Exception {
+
+                log.info("Check CApp un-deployment : " + cAppName);
+                boolean undeployed = true;
```

Comment:
I suppose we can use method "checkCAppDeployed" for this. WDYT?

Reply:
+1 updated